test/test_sendfile.rs

```diff
@@ -28,3 +37,109 @@ fn test_sendfile() {
     close(rd).unwrap();
     close(wr).unwrap();
 }
+
+#[cfg(target_os = "freebsd")]
+#[test]
+fn test_sendfile_freebsd() {
+    // Declare the content
+    let header_strings = vec!["HTTP/1.1 200 OK\n", "Content-Type: text/plain\n", "\n"];
+    let body = "Xabcdef123456";
+    let body_offset = 1;
+    let trailer_strings = vec!["\n", "Served by Make Believe\n"];
+
+    // Write the body to a file
+    let mut tmp = tempfile().unwrap();
+    tmp.write_all(body.as_bytes()).unwrap();
+
+    // Prepare headers and trailers for sendfile
+    let headers: Vec<IoVec<&[u8]>> = header_strings
```

Comment:
Would it make sense for the API to accept a slice of slices instead of a slice of `IoVec`s?  That would simplify the user's code.  In fact, if you did this, then `SendfileHeaderTrailer` wouldn't even need to be public.

Reply:
How about a slice of byte slices? I feel like it makes the API even less obtuse.